AutoDraft/NHL_API.py

```diff
@@ -0,0 +1,499 @@
+    # returns game-by-game stats for given list of players across given seasons (can specify sspecific stats for smaller returns)
+    multiplayer_df = pd.DataFrame() # TODO: add a progress bar
+    for player_id in player_id_list: 
+        player_name, player_position = get_player_name_position(player_id)
+        if player_position == 'G': continue # don't include goalies
+        if len(season_id_list) == 1: # handle single or multiple season input lists
+            player_df = augment_player_dataframe(get_player_season_game_stats(player_id=player_id, season_id=season_id_list[0]))
+        else:
+            player_df = augment_player_dataframe(get_combined_player_season_game_stats(player_id=player_id, season_id_list=season_id_list))
+        if len(stat_list) != 0: # if a specific stat is given, only grab that
+            player_small_df = player_df.loc[:, ['date', 'gameNumber'] + stat_list] # keep the useful indices
+        else:
+            player_small_df = player_df # grab it all otherwise
+        player_small_df.reset_index(drop=True, inplace=True) # get rid of messy index
+        try:
+            player_small_df.insert(0, 'name', [player_name for _ in range(len(player_small_df))])
+        except ValueError: # still don't know why this is thrown sometimes
+            st.dataframe(player_small_df)
+            player_small_df.insert(0, 'errorName', [player_name for _ in range(len(player_small_df))])
+        # player_small_df.set_index('gameNumber', inplace=True)
+        # player_small_df.rename(columns={stat: player_name}, inplace=True)
+        multiplayer_df = pd.concat([multiplayer_df, player_small_df], axis=0)
+        multiplayer_df.to_csv('./data/assemble_multiplayer_stat_dataframe_TEMP.csv') # export temporary state
+    multiplayer_df.reset_index(drop=True, inplace=True)
+    if shape == 'rows': # TODO: fix this when required to feed data in a row-per-player fashion
+        multiplayer_df = multiplayer_df.transpose()
+        # multiplayer_df.set_index(player_id_list, inplace=True)
+        try:
+            multiplayer_df.insert(0, 'name', multiplayer_df.index)
+        except ValueError:
+            multiplayer_df.insert(0, 'errorName', multiplayer_df.index)
+        multiplayer_df.insert(0, 'playerId', player_id_list)
+        multiplayer_df.set_index('playerId', inplace=True)
+    return multiplayer_df
+
+def get_all_rosters(season_id_list=[20152016, 20162017, 20172018, 20182019]): # returns roster of all players across a given list of seasons
+    teams_df = get_teams()
+    team_ids = teams_df.loc[:, 'id']
+    full_roster_df = pd.DataFrame()
+    for team_id in team_ids:
+        team_full_roster = merge_team_rosters(team_id=team_id, season_id_list=season_id_list) # TODO: save player id's
+        full_roster_df = pd.concat([full_roster_df, team_full_roster])
+    return full_roster_df
+
+## ARIMA modelling
+def calculate_errors(residuals):
```

Comment:
if you feel like you have the time to organize, you might want to move non-NHL API functions to a different module (.py file). there's a lot going on in this file.

Reply:
Separated modules into individual files and created a package folder.